util/repeatable_thread_test.cc

```diff
@@ -54,13 +54,30 @@ TEST_F(RepeatableThreadTest, TimedTest) {
 TEST_F(RepeatableThreadTest, MockEnvTest) {
   constexpr uint64_t kSecond = 1000000;  // 1s = 1000000us
   constexpr int kIteration = 3;
-  mock_env_->set_current_time(0);  // in seconds
+  rocksdb::Env* env = rocksdb::Env::Default();
+  // Obtain the current (real) time in seconds and add 10 extra seconds to
+  // ensure that RepeatableThread::wait invokes TimedWait with a time greater
+  // than (real) current time. This is to prevent the TimedWait function from
+  // returning immediately without sleeping and releasing the mutex on certain
+  // platforms, e.g. OS X. If TimedWait returns immediately, the mutex will not
+  // be released, and RepeatableThread::TEST_WaitForRun never has a chance to
+  // execute the callback which, in this case, updates the result returned by
+  // mock_env->NowMicros. Consequently, RepeatableThread::wait cannot break out
+  // of the loop, causing test to hang.
+  // The extra 10 seconds is a best-effort approach because there seems no
+  // reliable and deterministic way to provide the aforementioned guarantee. By
+  // the time RepeatableThread::wait is called, it is no guarantee that the
+  // delay + mock_env->NowMicros will be greater than the current real time.
+  // However, 10 seconds should be sufficient in most cases.
+  uint64_t now_seconds = env->NowMicros() / kSecond + 10;
```

Comment:
It seems to me the extra 10 seconds delay won't actually delay the test? If so why not use an even larger delay?

Reply:
True, it won't delay the actual test. Will consider a larger value, which is still a best-effort, but should be sufficient.